sqlitecluster/SQLite.cpp

```diff
@@ -258,7 +258,16 @@ int SQLite::_sqliteWALCallback(void* data, sqlite3* db, const char* dbName, int
             SINFO("[checkpoint] Ready for complete checkpoint but skipping because less than 100 commits since last complete checkpoint.");
             return SQLITE_OK;
         }
-        // If we get here, then full checkpoints are enabled, and we have enough pages in the WAL file to perform one.
+
+        int dbInUse = 0;
+        int useCheckResult = sqlite3_file_control(db, "main", SQLITE_FCNTL_EXTERNAL_READER, (void*)&dbInUse);
```

Comment:
Just want to rubber 🦆 to make sure I follow this stuff.

`(void*)` is casting `&dbInUse` to a "void pointer" which is a pointer than can essentially be any type, this is required by the `xFileControl` method that `sqlite3_file_control` calls since `xFileControl` expects a pointer as it's 3rd argument.

by casting a reference to `dbInUse` (ie: `&dbInUse`) to the void pointer it allows the `sqlite3_file_control` method and the underlying `xFileControl` method that gets called to "save" the return value of `xFileControl` in the memory address of `dbInUse` so that when we check `dbInUse` on the next line it's properly updated to 1 or left as 0 depending on the outcome of `xFileControl` with the opcode `SQLITE_FCNTL_EXTERNAL_READER` check.

read these things to get _some_ understanding https://www.sqlite.org/c3ref/file_control.html, https://www.sqlite.org/c3ref/io_methods.html, https://www.geeksforgeeks.org/void-pointer-c-cpp/, https://stackoverflow.com/questions/16866333/what-is-void-casting-used-for